concordium-std-derive/src/lib.rs

```diff
@@ -1270,14 +1270,14 @@ fn find_length_attribute(attributes: &[syn::Attribute]) -> syn::Result<Option<u3
 /// string.
 fn find_state_parameter_attribute(
     attributes: &[syn::Attribute],
-) -> syn::Result<Option<syn::Ident>> {
+) -> syn::Result<Option<syn::TypePath>> {
     let value = match find_attribute_value(attributes, false, "state_parameter")? {
         Some(v) => v,
         None => return Ok(None),
     };
 
     match value {
-        syn::Lit::Str(value) => Ok(Some(syn::Ident::new(&value.value(), value.span()))),
+        syn::Lit::Str(value) => Ok(Some(value.parse()?)),
```

Comment:
It would be better to provide a nicer error message to the user.
```
Some(value.parse().map_errr(|e| syn::Error::new(value.span(), "state_parameter attribute value is not a valid type")?)
```
or some such that talks directly about the `state_parameter`.

Reply:
Sure. Added more descriptive error message and tweaked one of non-string branch.